src/vs/workbench/browser/editor.ts

```diff
@@ -8,13 +8,16 @@ import { SyncDescriptor } from 'vs/platform/instantiation/common/descriptors';
 import { Registry } from 'vs/platform/registry/common/platform';
 import { BaseEditor } from 'vs/workbench/browser/parts/editor/baseEditor';
 import { IConstructorSignature0, IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { URI } from 'vs/base/common/uri';
```

Comment:
@jrieken @bpasero Please take a look the changes to the Editor registry and the EditorInput specifically.